File: app/src/main/java/mx/dev/franco/automusictagfixer/ui/BaseViewModelRoundedBottomSheetDialogFragment.java
package mx.dev.franco.automusictagfixer.ui;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import javax.inject.Inject;

import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.AndroidSupportInjection;
import dagger.android.support.HasSupportFragmentInjector;

public abstract class BaseViewModelRoundedBottomSheetDialogFragment<ViewModel> extends BaseRoundedBottomSheetDialogFragment
        implements HasSupportFragmentInjector {

    @Inject
    protected DispatchingAndroidInjector<Fragment> fragmentDispatchingAndroidInjector;
    @Inject
    protected AndroidViewModelFactory androidViewModelFactory;
    protected ViewModel mViewModel;

    /**
     * Called when a fragment is first attached to its context.
     * {@link #onCreate(Bundle)} will be called after this.
     */
    @Override
    public void onAttach(Context context) {
        AndroidSupportInjection.inject(this);
        super.onAttach(context);
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mViewModel = getViewModel();
    }

    protected abstract ViewModel getViewModel();

    protected void loading(boolean isLoading) {}
}

File: app/src/main/java/mx/dev/franco/automusictagfixer/ui/BaseViewModelRoundedBottomSheetDialogFragment.kt
package mx.dev.franco.automusictagfixer.ui

import android.content.Context
import android.os.Bundle
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasAndroidInjector
import dagger.android.support.AndroidSupportInjection
import javax.inject.Inject

abstract class BaseViewModelRoundedBottomSheetDialogFragment<ViewModel: Any> :
    BaseRoundedBottomSheetDialogFragment(), HasAndroidInjector{
    @Inject
    lateinit var fragmentDispatchingAndroidInjector: DispatchingAndroidInjector<Any>

    @Inject
    lateinit var androidViewModelFactory: AndroidViewModelFactory
    protected lateinit var vm: ViewModel

    /**
     * Called when a fragment is first attached to its context.
     * [.onCreate] will be called after this.
     */
    override fun onAttach(context: Context) {
        AndroidSupportInjection.inject(this)
        super.onAttach(context)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        vm = getViewModel()
    }

    abstract fun getViewModel(): ViewModel
    protected fun loading(isLoading: Boolean) {}

    override fun androidInjector(): AndroidInjector<Any> = fragmentDispatchingAndroidInjector
}